frontend/src/firebase/actions.ts

```diff
@@ -152,21 +139,51 @@ export const removeAllForks = (taskId: string): void => {
   batch.commit().then(ignore);
 };
 
+type EditType = 'EDITING_MASTER_TEMPLATE' | 'EDITING_ONE_TIME_TASK' | 'FORKING_MASTER_TEMPLATE';
+
 export const editTaskWithDiff = (
   taskId: string,
-  forMasterTemplate: boolean,
+  editType: EditType,
   { mainTaskEdits, subTaskCreations, subTaskEdits, subTaskDeletions }: Diff,
 ): void => {
   const batch = db().batch();
-  if (forMasterTemplate) {
+  let updateTaskId = taskId;
+  if (editType === 'FORKING_MASTER_TEMPLATE') {
+    const { tasks, subTasks } = store.getState();
```

Comment:
I personally think it might be more efficient to apply the diff to a task fetch from the store to create a complete forked task, then call `addTask` above to actually add it. It has the added benefit of reusing the undo logic